src/main/java/com/redhat/devtools/intellij/quarkus/lsp4ij/LanguageServerWrapper.java

```diff
@@ -724,6 +724,10 @@ public String getLanguageId(Language language) {
     }
 
     void registerCapability(RegistrationParams params) {
+        // no-op if initializeFuture is null or already done
+        if (initializeFuture == null || initializeFuture.isDone()) {
```

Comment:
I don't think we should skip if the initializeFuture is done as it may lead to capabilities not being registered if this method is called after initialization phase is completed